### ISSUE TITLE
Assertion violation at solvers/funs/fun_solver.c:2660

### ISSUE THREAD

**Issue by rainoftime** [SRI-CSL/yices2#294]
Hi, for the following formula
~~~~
(set-logic QF_ALIA)
(declare-fun _substvar_1773_ () (Array Bool Bool))
(declare-const v16 Bool)
(declare-const arr-6085764253233397621_6085764253233397621-0 (Array Bool Bool))
(declare-const arr-6085764253233397621_6085764253233397621-2 (Array Bool Bool))
(declare-const arr--8474158384260862814_6085764253233397621-0 (Array (Array Bool Bool) Bool))
(assert (= (select (store (store arr-6085764253233397621_6085764253233397621-0 false false) true true) v16) true true true (select (store (store arr--8474158384260862814_6085764253233397621-0 arr-6085764253233397621_6085764253233397621-2 (select arr-6085764253233397621_6085764253233397621-0 true)) _substvar_1773_ true) (store (store arr-6085764253233397621_6085764253233397621-0 false false) true true)) true true true true true))
(assert (select arr--8474158384260862814_6085764253233397621-0 (store arr-6085764253233397621_6085764253233397621-0 v16 false)))
(check-sat)
~~~~

yices e1c479eb96f4
~~~~
yices_smt2: solvers/funs/fun_solver.c:2660: equal_mappings: Assertion `n <= card && m <= card && dv != (2147483647) && dw != (2147483647)' failed.
~~~~

**Comment by rainoftime** [comment]
Related issue: https://github.com/SRI-CSL/yices2/issues/252
(yices e1c479e works well for that instance)

**Comment by BrunoDutertre** [comment]
Fixed in 24e4b0d570caf6d9afdbe3dcbb65d4af0f994a17
